src/CalcViewModel/Common/CopyPasteManager.cpp

```diff
@@ -449,33 +448,37 @@ bool CopyPasteManager::TryOperandToULL(const wstring& operand, int numberBase, u
     {
         // Do nothing
     }
+    catch(...)
+    {
+        // Do nothing
+    }
 
     return false;
 }
 
 size_t CopyPasteManager::OperandLength(wstring operand, ViewMode mode, CategoryGroupType modeType, int programmerNumberBase)
 {
-    size_t len = 0;
     if (mode == ViewMode::Standard || mode == ViewMode::Scientific)
     {
-        len = StandardScientificOperandLength(operand);
+        return StandardScientificOperandLength(operand);
     }
     else if (mode == ViewMode::Programmer)
     {
-        len = ProgrammerOperandLength(operand, programmerNumberBase);
+        return ProgrammerOperandLength(operand, programmerNumberBase);
     }
     else if (modeType == CategoryGroupType::Converter)
     {
-        len = operand.length();
+        return operand.length();
     }
 
-    return len;
+    return 0;
 }
 
 size_t CopyPasteManager::StandardScientificOperandLength(wstring operand)
 {
+    // Loop until a decimal is found or if the end is reached
     bool hasDecimal = false;
-    for (size_t i = 0; i < operand.length(); i++)
+    for (size_t i = 0; i < operand.length() && !hasDecimal; i++)
```

Comment:
`hasDecimal = operand.find('.') !=std::string::npos;` ?

Reply:
Change applied in latest commit to this branch.